pyqtorch/noise.py

```diff
@@ -0,0 +1,356 @@
+
+    @property
+    def kraus_operator(self) -> list[Tensor]:
+        return self.kraus
+
+    def unitary(self, values: dict[str, Tensor] | Tensor = dict()) -> list[Tensor]:
+        # Since PyQ expects tensor.Size = [2**n_qubits, 2**n_qubits,batch_size].
+        return [kraus_op.unsqueeze(2) for kraus_op in self.kraus]
+
+    def dagger(self, values: dict[str, Tensor] | Tensor = dict()) -> list[Tensor]:
+        return [_dagger(kraus_op) for kraus_op in self.unitary(values)]
+
+    def forward(self, state: Tensor, values: dict[str, Tensor] | Tensor = dict()) -> Tensor:
+        """
+        Applies a noisy quantum channel on the input state.
+        The evolution is represented as a sum of Kraus operators:
+        .. math::
+            S(\\rho) = \\sum_i K_i \\rho K_i^\\dagger,
+
+        Each Kraus operator in the `kraus_list` is applied to the input state, and the result
+        is accumulated to obtain the evolved state.
+
+        Args:
+            state (Tensor): Input quantum state represented as a tensor.
+
+        Returns:
+            Tensor: Quantum state as a density matrix after evolution.
+
+        Raises:
+            TypeError: If the input `state` or `kraus_list` is not a Tensor.
+        """
+        if not isinstance(state, Tensor):
+            raise TypeError("The input must be a Tensor")
+        if not isinstance(state, DensityMatrix):
+            state = density_mat(state)
+        rho_evols: list[Tensor] = []
+        for kraus_unitary, kraus_dagger in zip(self.unitary(values), self.dagger(values)):
+            rho_evol: Tensor = operator_product(
+                kraus_unitary, operator_product(state, kraus_dagger, self.target), self.target
+            )
+            rho_evols.append(rho_evol)
+        rho_final: Tensor = torch.stack(rho_evols, dim=0)
+        rho_final = torch.sum(rho_final, dim=0)
+        return rho_final
+
+    @property
+    def device(self) -> torch.device:
+        return self._device
+
+    def to(self, device: torch.device) -> Noise:
+        super().to(device)
+        self._device = device
+        return self
+
+
+class BitFlip(Noise):
+    """
+    Initialize the BitFlip gate.
+
+    The bit flip channel is defined as:
+
+    .. math::
+        \\rho \\Rightarrow (1-p) \\rho + p X \\rho X^{\\dagger}
+
+    Args:
+        target (int): The index of the qubit being affected by the noise.
+        probability (float): The probability of a bit flip error.
+
+    Raises:
+        TypeError: If the probability value is not a float.
+    """
+
+    def __init__(
+        self,
+        target: int,
+        probability: float,
+    ):
+        if probability > 1.0 or probability < 0.0:
+            raise ValueError("The probability value is not a correct probability")
+        K0: Tensor = sqrt(1.0 - probability) * IMAT
+        K1: Tensor = sqrt(probability) * XMAT
+        Kraus_Bitflip: list[Tensor] = [K0, K1]
```

Comment:
```suggestion
        kraus_bitflip: list[Tensor] = [K0, K1]
```

Reply:
Changed all kraus variable name in lowercase.